lib/src/dartdoc_options.dart

```diff
@@ -56,6 +56,65 @@ class DartdocFileMissing extends DartdocOptionError {
   DartdocFileMissing(String details) : super(details);
 }
 
+class CategoryDefinition {
+  /// Internal name of the category.
+  final String name;
+
+  /// Displayed name of the category in docs, or null if there is none.
+  final String _displayName;
+
+  /// Canonical path of the markdown file used to document this category
+  /// (or null if undocumented).
+  final String documentationMarkdown;
+  CategoryDefinition(this.name, this._displayName, this.documentationMarkdown);
```

Comment:
I'm curious, what situation? I don't have a strong preference here. I prefer named params for optional values myself because I think it makes call sites more readable, and conveys intend about the param being optional.